[vrtnu] Fix extractor

bergoid opened the pull request:
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

<!--

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible

-->

This fixes the VRTNU extractor after the recent website revamp of https://www.vrt.be/vrtnu/

Fixes #4908


<details open><summary>Template</summary> <!-- OPEN is intentional -->

<!--

# PLEASE FOLLOW THE GUIDE BELOW

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes `[ ]` relevant to your *pull request* (like [x])
- Use *Preview* tab to see how your *pull request* will actually look like

-->

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

pateik commented:
Hi,

Thanks for your patch

I see your regexp on line 217 containing (?P<year>[0-9]{4})

This does seem to catch a lot of url's,  however not all of them have a strict 4 digit year in them.
Look at the URL https://www.vrt.be/vrtnu/a-z/de-ideale-wereld/2022-nj/de-ideale-wereld-d20220913/
With your current patch, this url is not recognized, is it possible to fix this?

jeroenj commented:
@bergoid as far as I can tell all content on VRT MAX (new brand name of VRT Nu) is DRM-protected. So even if downloading is fixed it won't be any useful? Or are there still DRM-free videos available on VRT MAX?

bergoid commented:
> 

VRT NWS programs (het journaal, de afspraak, terzake, de zevende dag) are not DRM-protected AFAIK.

bergoid commented:
> not all of them have a strict 4 digit year in them.

@pateik thanks, I didn't know that. I will fix it.

jeroenj commented:
Indeed, you're right. I've tested with https://www.vrt.be/vrtnu/a-z/vrt-nws-journaal/2022/vrt-nws-journaal-vrt-nws-journaal-19u-20220915/ and that indeed works. 👍

mrBliss commented:
Thanks for fixing this @bergoid :+1:

> > not all of them have a strict 4 digit year in them.
> 
> @pateik thanks, I didn't know that. I will fix it.

Another example: https://www.vrt.be/vrtnu/a-z/down-the-road/5/down-the-road-s5a1/
Actually, a lot of shows use this numbering scheme for their seasons.

Any reason you changed this line? You don't seem to be using the regex match groups. If you revert the change on that line, everything seems to work again.

bergoid commented:
> If you revert the change on that line, everything seems to work again.

@mrBliss, I've reverted the regex.

> I also noticed that the season and episode number are no longer extracted. That's unfortunate, as I rely on them all the time (in --output).

A number of extra fields for the output template are now extracted. Among them: series/season/episode info.

pukkandan commented:
Add back all deleted tests (even the ones from deleted extractors)

Ottaviocr commented:
I have this issue with today's VRT Journaal:

<pre>
[debug] Command-line config: ['-vU', '-R', 'infinite', '--fragment-retries', 'infinite', '--restrict-filenames', '--progress', '--downloader=aria2c', '--postprocessor-args', 'thread:1', '--abort-on-error', 'https://www.vrt.be/vrtnu/a-z/vrt-nws-journaal/2022/vrt-nws-journaal-vrt-nws-journaal-13u-20221012/']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.10.04 [4e0511f] (zip)
[debug] Python 3.9.2 (CPython 64bit) - Linux-5.18.0-0.deb11.3-amd64-x86_64-with-glibc2.33 (glibc 2.33)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 4.3.4-0, ffprobe 4.3.4-0
[debug] Optional libraries: Cryptodome-3.9.7, brotli-1.0.9, certifi-2020.06.20, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1690 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.10.04, Current version: 2022.10.04
yt-dlp is up to date (2022.10.04)
[debug] [VrtNU] Extracting URL: https://www.vrt.be/vrtnu/a-z/vrt-nws-journaal/2022/vrt-nws-journaal-vrt-nws-journaal-13u-20221012/
[VrtNU] vrt-nws-journaal-vrt-nws-journaal-13u-20221012: Downloading webpage
ERROR: [VrtNU] vrt-nws-journaal-vrt-nws-journaal-13u-20221012: Unable to extract media element; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 672, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/canvas.py", line 317, in _real_extract
    attrs = extract_attributes(self._search_regex(
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1224, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
</pre>

jeroenj commented on yt_dlp/extractor/vrt.py:
```diff
@@ -1,82 +1,216 @@
-            'duration': 115.17,
+            'duration': 41.05,
+            'release_date': '20220908',
+            'description': 'md5:816dcd9e3be706b16e4e32e3f723a5a1',
```
This seems to have changed:

```suggestion
            'description': 'md5:48f8b2cddc819ddb2d21455be1ace6ef',
```

jeroenj commented:
> Add back all deleted tests (even the ones from deleted extractors)

@pukkandan what would be the point of this? I just had a look and all tests from deleted extractors would be of no use as most of the domains no longer resolve or the videos are no longer available. Since the extractors are removed as part of this PR there's no point in keeping those tests, right?

As for the `https://www.vrt.be/vrtnu` related tests that have been removed: those were marked to be skipped in the old code. And in the new code they'd not be of much use either as both videos are DRM-protected.

The test that exists now is a good example of what could be downloaded with this extractor. As far as I can tell only DRM-free files are available for unregistered users.

jeroenj commented on yt_dlp/extractor/vrt.py:
```diff
@@ -1,82 +1,216 @@
+                    'format_id': format_type,
+                    'url': format_url,
+                })
+        self._sort_formats(formats)
```
Once this branch gets rebased against master this would no longer be needed ([source](https://github.com/yt-dlp/yt-dlp/commit/9f14daf22b4080ae1531a772ee7574959af4e2fa)):

```suggestion
```

pukkandan commented:
I did not realize the sites were dead

pukkandan commented:
@jeroenj I didn't originally realize you are not OP. Considering they haven't replied for months, I don't think there is any point in a review. Pls open a new PR if you wish to do so

jeroenj commented:
@pukkandan I've opened https://github.com/yt-dlp/yt-dlp/pull/6244 which includes the initial changes of this MR and the fixes mentioned here on top of it. I've also replaced the test by a working URL.

bashonly commented:
Superseded by #6244 and closed by 1a7dcca378e80a387923ee05c250d8ba122441c6
